server/src/main/java/org/opensearch/common/time/JavaDateFormatter.java

```diff
@@ -150,7 +151,7 @@ JavaDateFormatter getRoundupParser() {
         if (parsers.length == 0) {
             this.parsers = Collections.singletonList(printer);
         } else {
-            this.parsers = new CopyOnWriteArrayList<>(parsers);
+            this.parsers = new ArrayList<>(Arrays.asList(parsers));
```

Comment:
My apologies, since we do not modify parsers anymore:

```suggestion
            this.parsers = Arrays.asList(parsers);
```

is good